src/group_theory/perm/cycles.lean

```diff
@@ -337,34 +345,51 @@ else let ⟨m, hm₁, hm₂, hm₃⟩ := cycle_factors_aux l ((cycle_of f x)⁻
         hm₃⟩⟩
 
 /-- Factors a permutation `f` into a list of disjoint cyclic permutations that multiply to `f`. -/
-noncomputable def cycle_factors [fintype α] [linear_order α] (f : perm α) :
+def cycle_factors [fintype α] [linear_order α] (f : perm α) :
   {l : list (perm α) // l.prod = f ∧ (∀ g ∈ l, is_cycle g) ∧ l.pairwise disjoint} :=
 cycle_factors_aux (univ.sort (≤)) f (λ _ _, (mem_sort _).2 (mem_univ _))
 
+/-- Factors a permutation `f` into a list of disjoint cyclic permutations that multiply to `f`,
+  without a linear order. -/
+def trunc_cycle_factors [fintype α] (f : perm α) :
+  trunc {l : list (perm α) // l.prod = f ∧ (∀ g ∈ l, is_cycle g) ∧ l.pairwise disjoint} :=
+quotient.rec_on_subsingleton (@univ α _).1
+  (λ l h, trunc.mk (cycle_factors_aux l f h))
+  (show ∀ x, f x ≠ x → x ∈ (@univ α _).1, from λ _ _, mem_univ _)
+
+lemma induction_on_cycles [fintype β] (P : perm β → Prop) (base_one : P 1)
+  (base_cycles : ∀ σ : perm β, σ.is_cycle → P σ)
+  (induction_disjoint : ∀ σ τ : perm β, disjoint σ τ → P σ → P τ → P (σ * τ)) (σ : perm β) :
+  P σ :=
```

Comment:
This is an `induction` lemma, not an `induction_on` lemma - the difference is whether `(σ : perm β)` comes before the base case (`_on`) or last.

Reply:
So is `induction_cycles` an acceptable name, or perhaps `cycle_induction`?

Reply:
Or just move `(σ : perm β)` to before `base_one`?